libraries/Microsoft.Bot.Builder.Ai/LocaleConverterMiddleware.cs

```diff
@@ -28,16 +30,28 @@ public LocaleConverterMiddleware(string fromLocale, string toLocale)
             {
                 if (!String.IsNullOrWhiteSpace(message.Text))
                 {
-                    await ConvertLocaleMessageAsync(message);
+                    string fromLocale = _getUserLocale(context);
+                    ((BotContext)context)["LocaleConversionOriginalMessage"] = message.Text;
+                    await ConvertLocaleMessageAsync(message, fromLocale);
+                    var localeWasChanged = await _setUserLocale(context);
+                    if (!localeWasChanged)
+                    {   // if what the user said wasn't a directive to change the locale (or that directive failed), continue the pipeline
+                        await next();
+                    }
                 }
             }
             await next().ConfigureAwait(false);
         }
 
+        public static string GetOriginalMessage(IBotContext context)
+        {
+            string message =   ((BotContext)context)["LocaleConversionOriginalMessage"] as string;
```

Comment:
I might suggest this is scoped to a Translation namespace. Polluting the global namespace with this probably won't work as we scale up.